editoast/src/models/detector_layer.rs

```diff
@@ -0,0 +1,134 @@
+    ) -> Result<(), Error> {
+        if obj_ids.is_empty() {
+            return Ok(());
+        }
+        let obj_ids: Vec<String> = obj_ids.into_iter().collect();
+
+        sql_query(include_str!("sql/insert_update_detector_layer.sql"))
+            .bind::<Integer, _>(infra)
+            .bind::<Array<Text>, _>(&obj_ids)
+            .execute(conn)?;
+        Ok(())
+    }
+
+    pub fn delete_list(
+        conn: &PgConnection,
+        infra: i32,
+        obj_ids: HashSet<String>,
+    ) -> Result<(), Error> {
+        if obj_ids.is_empty() {
+            return Ok(());
+        }
+
+        let obj_ids: Vec<String> = obj_ids.into_iter().collect();
+
+        sql_query("DELETE FROM osrd_infra_detectorlayer WHERE infra_id = $1 AND obj_id = ANY($2)")
+            .bind::<Integer, _>(infra)
+            .bind::<Array<Text>, _>(&obj_ids)
+            .execute(conn)?;
+
+        Ok(())
+    }
+
+    fn fill_detector_track_refs(
+        infra_cache: &InfraCache,
+        track_id: &String,
+        results: &mut HashSet<String>,
+    ) {
+        infra_cache
+            .get_track_refs_type(track_id, ObjectType::Detector)
+            .iter()
+            .for_each(|obj_ref| {
+                results.insert(obj_ref.obj_id.clone());
+            });
+    }
+
+    /// Search and update all detectors that needs to be refreshed given a list of operation.
+    pub fn update(
+        conn: &PgConnection,
+        infra: i32,
+        operations: &Vec<OperationResult>,
+        infra_cache: &InfraCache,
+        invalid_zone: &InvalidationZone,
+        chartos_config: &ChartosConfig,
+    ) -> Result<(), Error> {
+        let mut update_obj_ids = HashSet::new();
+        let mut delete_obj_ids = HashSet::new();
+        for op in operations {
+            match op {
+                OperationResult::Create(RailjsonObject::TrackSection { railjson })
+                | OperationResult::Update(RailjsonObject::TrackSection { railjson }) => {
+                    Self::fill_detector_track_refs(infra_cache, &railjson.id, &mut update_obj_ids)
+                }
+                OperationResult::Create(RailjsonObject::Detector { railjson })
+                | OperationResult::Update(RailjsonObject::Detector { railjson }) => {
+                    update_obj_ids.insert(railjson.id.clone());
+                }
+                OperationResult::Delete(ObjectRef {
+                    obj_type: ObjectType::Detector,
+                    obj_id: detector_id,
+                }) => {
+                    delete_obj_ids.insert(detector_id.clone());
+                }
+                _ => (),
+            }
+        }
+
+        if update_obj_ids.is_empty() && delete_obj_ids.is_empty() {
+            // No update needed
+            return Ok(());
+        }
+
+        Self::delete_list(conn, infra, delete_obj_ids)?;
+        Self::insert_update_list(conn, infra, update_obj_ids)?;
+
+        if check_bbox_bound(invalid_zone) {
```

Comment:
You shouldn't check this here.
If the bounding box is invalid then it makes not sense to update any layers.
So the check can be made in `src/views/infra.rs`